File: Implement Atoi.java
class Solution {
    public int myAtoi(String s) {
        // code here
        s = s.trim();
        if (s.isEmpty()) {
            return 0;
        }
        int sign = 1;
        int i = 0;
        if (s.charAt(0) == '-') {
            sign = -1;
            i++;
        } else if (s.charAt(0) == '+') {
            i++;
        }
        long b = 0; 
        while (i < s.length()) {
            char c = s.charAt(i);
            if (!Character.isDigit(c)) {
                break;
            }
            b = b * 10 + (c - '0');
            if (sign * b >= Integer.MAX_VALUE) return Integer.MAX_VALUE;
            if (sign * b <= Integer.MIN_VALUE) return Integer.MIN_VALUE;
            i++;
        }
        
        return (int)(sign * b);
    }
}

File: Implement Atoi.py
class Solution:
    def myAtoi(self, s):
        # code here
        s = s.strip()
        if not s:
            return 0
        a = -1 if s[0] == "-" else 1
        if s[0] in ["+","-"]:
            s = s[1:]
        b = 0
        for c in s:
            if not c.isdigit():
                break
            b = b * 10 + int(c)
        b *= a
        return max(-2**31, min(b, 2**31-1))
